test/multi_config/multi_config_linux_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build !windows
+
+package multi_config
+
+import (
+	"log"
+	"testing"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/awsservice"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+)
+
+const (
+	configOutputPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+	namespace        = "MultiConfigTest"
+)
+
+// Let the agent run for 2 minutes. This will give agent enough time to call server
+const agentRuntime = 2 * time.Minute
+
+var envMetaDataStrings = &(environment.MetaDataStrings{})
+
+func init() {
+	environment.RegisterEnvironmentMetaDataFlags(envMetaDataStrings)
+}
+
+func TestMultipleConfig(t *testing.T) {
+
+	agentConfigurations := []string{"resources/LinuxLogOnlyConfig.json", "resources/LinuxMemoryOnlyConfig.json"}
+
+	for index, agentConfig := range agentConfigurations {
+		common.CopyFile(agentConfig, configOutputPath)
+		log.Printf(configOutputPath)
+		if index == 0 {
+			common.StartAgent(configOutputPath, true, false)
+		} else {
+			common.StartAgentWithMultiConfig(configOutputPath, true, false)
+		}
+	}
+
+	time.Sleep(agentRuntime)
+	log.Printf("Agent has been running for : %s", agentRuntime.String())
+	common.StopAgent()
+
+	// test for cloud watch metrics
+	ec2InstanceId := awsservice.GetInstanceId()
+	expectedDimensions := []types.DimensionFilter{
+		types.DimensionFilter{
+			Name:  aws.String("InstanceId"),
+			Value: aws.String(ec2InstanceId),
+		},
+	}
+
+	expectedMetrics := []string{"mem_used_percent"}
+	for _, expectedMetric := range expectedMetrics {
+		awsservice.ValidateMetrics(t, expectedMetric, namespace, expectedDimensions)
+	}
```

Comment:
This test has a log config and a metric config, but we only validate the 1 memory metric.
Instead I think it would be better to also validate some log data.
Or if that is not possible, consider adding a few more configs for cpu, disk, net, etc... And after each added config check for the presence of the new metric.

1. Start with config for just mem
2. Verify mem metrics
3. Add config for cpu
4. Verify cpu metrics
5. Add config for ...
6. Verify ... metrics

I realize this might be more than what we tested in the past, but that is a good thing :)